packages/typegpu/tests/rawFn.test.ts

```diff
@@ -150,4 +150,54 @@ describe('tgpu.fn with raw string WGSL implementation', () => {
 
     expect(actual).toEqual(expected);
   });
+
+  it('adds output struct definition when resolving vertex functions', () => {
+    const vertexFunction = tgpu
+      .vertexFn(
+        { vertexIndex: builtin.vertexIndex },
+        { outPos: builtin.position },
+      )
+      .does(/* wgsl */ `(@builtin(vertex_index) vertexIndex: u32) -> VertexOutput {
+    var pos = array<vec2f, 6>(
+      vec2<f32>( 1,  1),
+      vec2<f32>( 1, -1),
+      vec2<f32>(-1, -1),
+      vec2<f32>( 1,  1),
+      vec2<f32>(-1, -1),
+      vec2<f32>(-1,  1)
+    );
+  
+    var output: VertexOutput;
+    output.outPos = vec4f(pos[vertexIndex], 0, 1);
+    return output;
+  }`)
+      .$name('vertex_fn');
+
+    expect(
+      tgpu.resolve({ input: [vertexFunction], names: 'strict' }),
+    ).toContain(`\
+struct vertex_fn_Output {
+  @builtin(position) outPos: vec4f,
+}`);
+  });
+
+  it('adds output struct definition when resolving fragment functions', () => {
+    const fragmentFunction = tgpu
+      .fragmentFn({ position: builtin.position }, { a: vec4f, b: vec4f })
+      .does(/* wgsl */ `(@builtin(position) position: vec4f) -> Output {
+    var out: Output;
+    out.a = vec4f(1.0);
+    out.b = vec4f(0.5);
+    return out;
+  }`)
+      .$name('fragment');
+
+    expect(
+      tgpu.resolve({ input: [fragmentFunction], names: 'strict' }),
+    ).toContain(`\
+struct fragment_Output {
+  @location(0) a: vec4f,
+  @location(1) b: vec4f,
+}`);
+  });
 });
```

Comment:
Test case I used locally
```suggestion

  it('does not add redundant struct definition', () => {
    const vertexFunction = tgpu
      .vertexFn({ vertexIndex: builtin.vertexIndex }, builtin.position)
      .does(/* wgsl */ `(@builtin(vertex_index) vertexIndex: u32) -> @builtin(position) vec4f {
    let pos = array<vec2f, 6>(
      vec2<f32>( 1,  1),
      vec2<f32>( 1, -1),
      vec2<f32>(-1, -1),
      vec2<f32>( 1,  1),
      vec2<f32>(-1, -1),
      vec2<f32>(-1,  1)
    );

    return vec4f(pos[vertexIndex], 0, 1);
  }`)
      .$name('vertex_fn');

    expect(
      tgpu.resolve({ input: [vertexFunction], names: 'strict' }),
    ).not.toContain('struct');
  });
});
```